### PR TITLE
[16.0][FIX] web_advanced_search: Fixed changes from code editor not propagated

### PR DESCRIPTION
**Pull request by chienandalu**
Propagate changes in the debug mode domain editor to the domain itself.

cc @Tecnativa TT51315

please review @CarlosRoca13 @carlosdauden

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]
Code review :+1:

File: web_advanced_search/static/src/js/DomainSelector.esm.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            return this._super(...arguments);
+        }
+        const rawDomain = e.currentTarget.value;
+        try {
+            Domain.prototype.stringToArray(rawDomain);
+        } catch (err) {
+            // If there is a syntax error, just ignore the change
+            this.displayNotification({
+                title: _t("Syntax error"),
+                message: _t("Domain not properly formed"),
```
Comment (pedrobaeza) [inline web_advanced_search/static/src/js/DomainSelector.esm.js:28]: For this to be exported for translation, we need to override http.py, isn't it?

Comment (chienandalu) [reply web_advanced_search/static/src/js/DomainSelector.esm.js:28]: That's for frontend

Comment (pedrobaeza) [reply web_advanced_search/static/src/js/DomainSelector.esm.js:28]: OK then

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Review by carlosdauden — COMMENTED** [review]
Ok to this PR.
Only comment that the "add new value" option does not work correctly, I understand that this problem is different from the one solved by this PR and that functionally it is not important because the only way to use the "in" operator is using the debug mode.

But as I have detected it in my functional review, I leave it commented.

image

**Pull request merged**